Tasks/intersection.js

```diff
@@ -1,25 +1,14 @@
 // Find an intersection of two dictionaries
-
-intersection = function intersection(object_1, object_2) {
-  first_keys = Object.keys(object_1);
-  {
-  }
-  for (attribute_name of first_keys) {
-    if (object_1[attribute_name] === object_2[attribute_name]) {
-      object_2[attribute_name] = object_1[attribute_name];
-      {
-      }
-    } else {
-      delete object_1[attribute_name];
-      {
-      }
+'use strict';
+const intersection = (object1, object2) => {
+  const object1Keys = Object.keys(object1);
+  const result = {};
+  for (const key of object1Keys) {
+    if (object2[key] === object1[key]) {
+      result[key] = object1[key];
```

Comment:
You can save `object[key]` to intermediate variable not to repeat execution of the same expression

Reply:
Thank you for finding my code`s disadvantages. I`ve got it, thus I`ve made new remastered commits with the following fixes. There is the link : https://github.com/HowProgrammingWorks/Exams/compare/master...SharpDevOps10:Exams:exams
P.S new commits are marked as "remastered " and made on Jan 25, 2023